api/src/components/additionalInformation/__tests__/createAdditionalInformation.test.ts

```diff
@@ -0,0 +1,114 @@
+        const createPayload = {
+            title: 'Example title',
+            url: 'https://jisc.ac.uk'
+        };
+        const createAdditionalInformation = await testUtils.agent
+            .post('/publication-versions/publication-problem-draft-v1/additional-information')
+            .query({ apiKey: '000000005' })
+            .send(createPayload);
+
+        expect(createAdditionalInformation.status).toEqual(200);
+        expect(createAdditionalInformation.body).toMatchObject(createPayload);
+    });
+
+    test('User cannot add additional information to a non-existent publication version', async () => {
+        const createAdditionalInformation = await testUtils.agent
+            .post('/publication-versions/made-up-publication-version-id/additional-information')
+            .query({ apiKey: '000000005' })
+            .send({
+                title: 'Example title',
+                url: 'https://jisc.ac.uk'
+            });
+
+        expect(createAdditionalInformation.status).toEqual(404);
+        expect(createAdditionalInformation.body.message).toEqual('This publication version does not exist');
+    });
+
+    test('User cannot add additional information to their LIVE publication version', async () => {
+        const createAdditionalInformation = await testUtils.agent
+            .post('/publication-versions/publication-problem-live-v1/additional-information')
+            .query({ apiKey: '123456789' })
+            .send({
+                title: 'Example title',
+                url: 'https://jisc.ac.uk'
+            });
+
+        expect(createAdditionalInformation.status).toEqual(403);
+        expect(createAdditionalInformation.body.message).toEqual(
+            'You can only add additional information to a draft publication'
+        );
+    });
+
+    test('User cannot add additional information to a DRAFT publication version they do not own', async () => {
+        const createAdditionalInformation = await testUtils.agent
+            .post('/publication-versions/publication-problem-draft-v1/additional-information')
+            .query({ apiKey: '123456789' })
+            .send({
+                title: 'Example title',
+                url: 'https://jisc.ac.uk'
+            });
+
+        expect(createAdditionalInformation.status).toEqual(403);
+        expect(createAdditionalInformation.body.message).toEqual(
+            'You cannot add additional information to a publication version you do not own'
+        );
+    });
+
+    test('User cannot create additional information without supplying a title', async () => {
+        const createAdditionalInformation = await testUtils.agent
+            .post('/publication-versions/publication-problem-draft-v1/additional-information')
+            .query({ apiKey: '000000005' })
+            .send({
+                url: 'https://jisc.ac.uk'
+            });
+
+        expect(createAdditionalInformation.status).toEqual(422);
```

Comment:
I'd argue this (and the other responses tested below) should be 400 responses rather than 422. While both are similar enough, and we could probably debate this for a while, 400 is much more common and "standard" nowadays. I had to google to remember what 422 meant, while most api users will instantly know what 400 means.

In my opinion, it's a tradeoff between being standards compliant (which is difficult with http) and keeping it simple. I'd always favour keeping it simple, especially when it's an internal api.

Reply:
Agree with this - just pushed a commit which changes our response code for schema validation failures from 422 to 400.